### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by sirex**
All tests pass, three are skipped. Also included my favorite `scripts/runtests.py` script.

With migrations, still there is no guarantee that hack with auth.User model will work, I will test that later, when authentication will be fixed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ bin/django bin/sassc: bin/buildout buildout.cfg $(wildcard config/*.cfg) $(wildc
 migrate: bin/django ; bin/django migrate
 
 reset_mysql:
-	mysql -e 'drop database manoseimas; create database manoseimas character set utf8 collate utf8_bin;'
+	mysql -e 'drop database if exists manoseimas; create database manoseimas character set utf8 collate utf8_bin;'
```
Comment (frgtn) [inline Makefile]: Good catch.

**Comment by frgtn** [comment]
Looks good to me.

**Comment by frgtn** [comment]
Apparently the user model hack didn't work and `runserver` dies with the following error:

```
django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:

ERRORS:
profiles.Profile.user: (fields.E301) Field defines a relation with the model 'auth.User', which has been swapped out.
	HINT: Update the relation to point at 'settings.AUTH_USER_MODEL'.
```

Also, I missed that runtests makes assumptions about location of parts files, which differ in Vagrant setup.

Since I cannot reopen this pull request, it's probably best to fix these issues and make a new one.

File: scripts/runtests.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+def run_coverage_report(args):
+    # Also see .coveragerc
+    return subprocess.call(['bin/coverage', 'report', '--show-missing'])
+
+
+def main(args=None):
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        'paths', nargs='*', default=['manoseimas', 'parts/django-sboard'],
```
Comment (frgtn) [inline scripts/runtests.py]: This should probably use `exportrecipe` to determine where `django-sboard`.

**Comment by sirex** [comment]
For the error, you have to update django-sboard too.

File: scripts/runtests.py (hunk 2 above)
Comment (sirex) [inline scripts/runtests.py]: Yes, it should, I will add it next week.

**Comment by frgtn** [comment]
Cool, I didn't try that. I think it's a good idea to pin django-sboard revision in buildout.cfg somewhere, like this (from `conf/env/vagrant.cfg`):
```
django-sboard = hg https://bitbucket.org/sirex/django-sboard rev=74877b530b1d1378835daf1bb2421e913452ddbf
```

**Comment by sirex** [comment]
There is an option [always-checkout](https://pypi.python.org/pypi/mr.developer), if it set to `true` repositories always updated on `bin/buildout`, but while I'm constantly changing it, I prefer the manual way.

Also, since `django-sboard` is not going to be a separate project, maybe it is worth to move it to manoseimas repository.

**Comment by frgtn** [comment]
I agree, if `django-sboard` isn't really an external dependency, it's better to move it into manoseimas repo.

`always-checkout` doesn't give you reproducible builds - if you checkout a different branch or an older commit it will still use latest rev of `django-sboard`, which might not be what you want. For fully reproducible builds you need to pin a revision explicitly.

**Comment by sirex** [comment]
Well, I think I'm too lazy to do the manual pinning each time I change `django-sboard`, which was many times. It should be automated. But anyway, since we going to move to the repo. And for the `couchdbkit`, maybe it is worth pinning, because it has much fewer changes.

**Pull request merged**